packages/text-input/src/types.ts

```diff
@@ -38,26 +38,41 @@ export const TextInputFontSize = {
 export type TextInputFontSize =
   typeof TextInputFontSize[keyof typeof TextInputFontSize];
 
-export interface BaseTextInputProps
-  extends HTMLElementProps<'input', HTMLInputElement> {
-  /**
-   * id associated with the TextInput component.
-   */
-  id?: string;
-
+interface AriaLabelProps {
   /**
    * Text shown in bold above the input element.
    *
    * Optional if `aria-labelledby` is provided
    */
-  label: string | null;
+  label?: string;
 
   /**
    * Screen-reader label element.
    *
    * Optional if `label` is provided
    */
-  ['aria-labelledby']: string;
+  ['aria-labelledby']?: string;
+}
+
+type AriaLabels = keyof AriaLabelProps;
+
+interface TextInputTypeProp {
+  /**
+   * The input type.
+   *
+   * Requires an `aria-label` if `"search"` is provided
```

Comment:
Aria-labelledby can be sufficient here as well. We don't necessarily need aria-label if search is provided.